### ISSUE TITLE
[Feature] RandomErasingOp (cutout) augmentation

### ISSUE THREAD

**Issue by brody-0125** [brody-0125/dart_tensor_preprocessing#42]
## Summary

Add `RandomErasingOp` that randomly erases a rectangular region of a tensor image, also known as "cutout" augmentation.

## Motivation

Random erasing is a widely used regularization technique that improves model robustness by occluding random parts of training images. It's particularly effective for object detection and recognition tasks.

## Reference Implementations

- **PyTorch**: `torchvision.transforms.RandomErasing(p=0.5, scale=(0.02, 0.33), ratio=(0.3, 3.3), value=0)` — [PyTorch docs](https://pytorch.org/vision/stable/generated/torchvision.transforms.RandomErasing.html)
- Paper: "Random Erasing Data Augmentation" (Zhong et al., 2020, AAAI)

## Verification

- ✅ `torchvision.transforms.RandomErasing` exists — VERIFIED (selects random rectangle and erases pixels)
- ✅ Feature is MISSING from dart_tensor_preprocessing — VERIFIED (searched for erasing, cutout, erase)

## Proposed API

```dart
final erase = RandomErasingOp(
  probability: 0.5,
  scaleRange: (0.02, 0.33),  // area fraction range
  ratioRange: (0.3, 3.3),    // aspect ratio range
  value: 0.0,                 // fill value (or 'random')
);
```

## Implementation Notes

- Randomly select area fraction and aspect ratio within specified ranges
- Calculate erasing rectangle dimensions from area and ratio
- Place rectangle at random position
- Fill with constant value or random noise
- Support 3D `[C, H, W]` and 4D `[N, C, H, W]`
- Optional seed for reproducibility
- In-place capable for efficiency

## ROI Assessment

**Medium-High** — Proven augmentation technique with demonstrated effectiveness. Simple to implement, good impact on model training robustness. Complements existing RandomCropOp.
